src/clients/stsService.js

```
@@ -0,0 +1,37 @@
+import logger from '../utils/logger.js';
+import { GetCallerIdentityCommand, STSClient } from '@aws-sdk/client-sts';
+
+// Factory function to create an STS client
+export const createAwsSTSClient = () => {
+  logger.debug('Creating AWS STSClient');
+  return new STSClient();
+};
+
+export class STSService {
+  constructor(stsClient) {
+    if (!stsClient) throw new Error('STSClient is required');
```

ESLint

[error] Expected { after 'if' condition.

(curly)